invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/src/deposit/components/CommunityHeader/CommunityHeader.js

```diff
@@ -64,47 +66,53 @@ class CommunityHeaderComponent extends Component {
-                    displaySelected
-                    record={record}
-                    trigger={
-                      <Button
-                        className="community-header-button"
-                        disabled={disableCommunitySelectionButton}
-                        onClick={() => this.setState({ modalOpen: true })}
-                        primary
-                        size="mini"
-                        name="setting"
-                        type="button"
-                        content={
-                          community
-                            ? i18next.t("Change")
-                            : i18next.t("Select a community")
-                        }
-                      />
-                    }
-                  />
-                  {community && (
-                    <Button
-                      basic
-                      size="mini"
-                      labelPosition="left"
-                      className="community-header-button ml-5"
-                      onClick={() => changeSelectedCommunity(null)}
-                      content={i18next.t("Remove")}
-                      icon="close"
-                      disabled={disableCommunitySelectionButton}
+                <Overridable id="InvenioRdmRecords.CommunityHeader.CommunityHeaderElement.container">
+                  <>
+                    <CommunitySelectionModal
+                      onCommunityChange={(community) => {
+                        changeSelectedCommunity(community);
+                        this.setState({ modalOpen: false });
+                      }}
+                      onModalChange={(value) => this.setState({ modalOpen: value })}
+                      modalOpen={modalOpen}
+                      chosenCommunity={community}
+                      displaySelected
+                      record={record}
+                      trigger={
+                        <Overridable id="InvenioRdmRecords.CommunityHeader.SelectCommunity.Button">
+                          <Button
+                            className="community-header-button"
+                            disabled={disableCommunitySelectionButton}
+                            onClick={() => this.setState({ modalOpen: true })}
+                            primary
+                            size="mini"
+                            name="setting"
+                            type="button"
+                            content={
+                              community
+                                ? i18next.t("Change")
+                                : i18next.t("Select a community")
+                            }
+                          />
+                        </Overridable>
+                      }
                     />
-                  )}
-                </>
+                    {community && (
+                      <Overridable id="InvenioRdmRecords.CommunityHeader.RemoveCommunity.Button">
```

Comment:
```suggestion
           <Overridable id="InvenioRdmRecords.CommunityHeader.RemoveCommunity.container" community={community}>
                    {community && (

```
passing the community as a prop to the overridable component will allow to control also the conditional statement on when this component is displayed (while overriding)